src/sources/notification_permissions.test.ts

```diff
@@ -1,10 +1,19 @@
 import { getBrowserMajorVersion, getBrowserVersion, isHeadlessChrome, isMobile, isWebKit } from '../../tests/utils'
-import { BotdError } from '../types'
+import { BotdError, BrowserEngineKind, BrowserKind } from '../types'
+import { getBrowserEngineKind, getBrowserKind, isDesktopWebKit } from '../utils/browser'
 import getNotificationPermissions from './notification_permissions'
 
 describe('Sources', () => {
   describe('notificaionPermissions', () => {
     it('returns an expected value or throws', async () => {
+      if (
+        getBrowserKind() === BrowserKind.Safari &&
+        getBrowserEngineKind() === BrowserEngineKind.Webkit &&
+        !isDesktopWebKit()
+      ) {
+        return
```

Comment:
I think it's better to check that the source returns `false` instead of skipping this case

Reply:
Also, the below checks for WebKit test for `isMobile()` which seems redundant after this change. I think we should update the check for `toBeRejectedWithError` instead of completely skipping the test for Safari on iOS.